client/components/amount-input/index.tsx

```diff
@@ -2,17 +2,30 @@
  * External dependencies
  */
 import React, { useCallback, useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
 import './style.scss';
 
-const AmountInput = ( {
+interface Props {
+	id: string;
+	prefix: string;
+	value: any;
+	placeholder: string;
+	help: string;
+	onChange: ( value: string ) => void;
+}
+
+const AmountInput: React.FunctionComponent< Props > = ( {
 	id,
 	prefix,
 	value,
 	placeholder,
 	help,
-	onChange = () => {},
+	onChange,
```

Comment:
Just to make sure: did we check that the `onChange` function is provided everywhere this component is used? 🤔 

Just want to make sure since there might be some JS code that's using this component and TS won't catch any errors when the usage is in JS.